### PR TITLE
PLAT-858: register plugins on auto upgrade

### PR DESCRIPTION
**Pull request by alecsavvy**
### Description
Adds `audius-cli` commands to register, deregister, and view registered plugins. 
Installs them after autoupgrade so long as a valid `docker-compose.{plugin name}.yml` exists in the disc prov dir.

### Snags
Assumes plugins only hook into discprov. This won't work for identity or storage services.

### CODE REVIEW

File: audius-cli [hunk 1]
```diff
@@ -859,5 +908,52 @@ def launch_notifications(ctx):
...
+        if not plugins:
+            return []
+    plugins = ",".join(plugins)
+    dotenv.set_key(override_env, REGISTERED_PLUGINS, plugins)
+    return
+
+
+@cli.command()
+@click.pass_context
+def plugins(ctx):
```
Comment (sabrina-kiam) [inline audius-cli:951]: i can't tell if its more clear to rename to `print_plugins` ?? ill leave up to you

**Review by sabrina-kiam — APPROVED** [review]
nice! a few nits otherwise this looks great

File: audius-cli [hunk 2]
```diff
@@ -859,5 +908,52 @@ def launch_notifications(ctx):
...
+    else:
+        return []
+    if name in plugins:
+        plugins.remove(name)
+        # last plugin removed
+        if not plugins:
+            return []
+    plugins = ",".join(plugins)
+    dotenv.set_key(override_env, REGISTERED_PLUGINS, plugins)
+    return
```
[the remark this answers is not in the document]
Comment (cheran-senthil) [reply audius-cli:946]: Never return anything on command functions, it can get confusing

**Review by cheran-senthil — CHANGES_REQUESTED** [review]
plugin_register -> register_plugin
plugin_deregister -> deregister_plugin
plugins -> list_plugins

or alternatively create `plugin` subcommand group and use
```sh
audius-cli plugin register
audius-cli plugin deregister
audius-cli plugin list
```

File: audius-cli (hunk 1 above)
Comment (alecsavvy) [reply audius-cli:951]: ah yeah i like that!

File: audius-cli (hunk 2 above)
[the remark this answers is not in the document]
Comment (alecsavvy) [reply audius-cli:946]: ah yeah, this started as a utility function and forgot to remove that

**Review by cheran-senthil — APPROVED** [review]
lgtm 👍

**Pull request merged**